scio-core/src/main/scala/com/spotify/scio/coders/CoderMaterializer.scala

```diff
@@ -35,8 +35,8 @@ object CoderMaterializer {
     coder match {
       case Beam(c) =>
         val nullableCoder = o.as(classOf[com.spotify.scio.options.ScioOptions]).getNullableCoders()
-        if (nullableCoder) NullableCoder.of(c)
-        else c
+        if (nullableCoder) WrappedBCoder.create(NullableCoder.of(c))
```

Comment:
@jto nice catch here! I found that KvCoder is not wrapped as well.  Maybe we can make this like 
```scala
  final def beam[T](r: CoderRegistry, o: PipelineOptions, coder: Coder[T]): BCoder[T] =
    WrappedBCoder.create {
         ...
    }
```